### PR TITLE
Port InclusiveKinematics reconstruction from Juggler

### PR DESCRIPTION
**Pull request by 2oba**
### Briefly, what does this PR introduce?
Port all InclusiveKinematics reconstruction algorithms from Juggler. This does not implement an electron finder and just uses a very ad-hoc electron finder. But, modularity above everything, we can write an electron finder later and feed it in here.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [X] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [X] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
NO
### Does this PR change default behavior?
NO

### CODE REVIEW

**Comment by DraTeots** [comment]
Hello! Looking at the code, shouldn't this PR be marked as a draft for now?

**Comment by wdconinc** [comment]
There is no CI check for DIS but I'll try to put one in another PR later this weekend.

**Review by DraTeots — APPROVED** [review]

**Comment by wdconinc** [comment]
```console
root [3] events->Scan("InclusiveKinematicsElectron.Q2:InclusiveKinematicsTruth.Q2")
***********************************************
*    Row   * Instance * Inclusive * Inclusive *
***********************************************
*        0 *        0 * 1.1372342 * 1.0909189 *
*        1 *        0 * 1.6724832 * 1.7164249 *
*        2 *        0 * 1.7320253 * 1.8345201 *
*        3 *        0 * 1.9497621 * 2.0091736 *
*        4 *        0 * 2.1081485 * 2.1205902 *
*        5 *        0 * 1.0558173 * 1.0399674 *
*        6 *        0 * 7.1425447 * 7.4604287 *
*        7 *        0 * 1.7359403 * 1.8598725 *
*        8 *        0 * 1.2875204 * 1.3207134 *
*        9 *        0 * 1.3967217 * 1.2877374 *
*       10 *        0 * 1.5751532 * 1.6150044 *
*       11 *        0 * 1.5031118 * 1.4772810 *
*       12 *        0 * 1.9913166 * 1.9352692 *
*       13 *        0 * 1.1622288 * 1.1194691 *
*       14 *        0 * 2.3158884 * 2.4200272 *
*       15 *        0 * 1.3911742 * 1.3647135 *
*       16 *        0 * 6.6320381 * 6.6487374 *
*       17 *        0 * 1.4039851 * 1.4092354 *
*       18 *        0 * 1.1211439 * 1.1058815 *
*       19 *        0 * 2.6301772 *  2.737463 *
*       20 *        0 * 2.8053386 * 2.8432242 *
*       21 *        0 * 1.0971818 * 1.1105676 *
*       22 *        0 * 1.3298379 * 1.3005757 *
*       23 *        0 * 1.0387001 *  1.034554 *
*       24 *        0 * 1.0160417 * 1.0811655 *
```

File: src/services/io/podio/JEventProcessorPODIO.cc [hunk 1]
```diff
@@ -103,6 +103,8 @@ JEventProcessorPODIO::JEventProcessorPODIO() {
             "ReconstructedChargedParticles",
             "ReconstructedChargedParticlesAssociations",
             "CentralTrackSegments",
+            "InclusiveKinematicsElectron",
+            "InclusiveKinematicsTruth",
```
Comment (wdconinc) [inline src/services/io/podio/JEventProcessorPODIO.cc:107]:
```suggestion
            "InclusiveKinematicsDA",
            "InclusiveKinematicsJB",
            "InclusiveKinematicsSigma",
            "InclusiveKinematicseSigma",
            "InclusiveKinematicsElectron",
            "InclusiveKinematicsTruth",
```

**Review by wdconinc — CHANGES_REQUESTED** [review]
This looks great, ready for merging as soon as we also add the DA, JB, etc methods to the output, as suggested.

**Review by wdconinc — APPROVED** [review]

**Pull request merged**